app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  
+  has_one :profile
+
+  validates :email, presence: true
+  validates :password, presence: true, confirmation: true
+  validates :password_confirmation, presence: true
+  validates :role, inclusion: { in: ['admin', 'customer'] }
```

Comment:
I noticed this is a string and I think this could be an enum, so you could have it like an integer and then use it like `category` in product